src/extensionFunctions.ts

```diff
@@ -896,4 +899,41 @@ export function sendTextToNamedOutputChannel(message: string, channelName?: stri
 	if (!channelName || channelName === DIDACT_OUTPUT_CHANNEL) {
 		didactOutputChannel.show();
 	}
-}
\ No newline at end of file
+}
+
+export async function copyFileTextToClipboard(uri: vscode.Uri) : Promise<void> {
+	let testForPrefix : string = uri.toString();
+	let testUri : vscode.Uri = uri;
+	if (testForPrefix && !testForPrefix.startsWith('didact://?')) {
+		testForPrefix = 'didact://?&' + testForPrefix; // preface so we can use the existing method to parse for path
```

Comment:
this sounds like a trick that will be hard to remember and maintain. A specific method should be extracted

Reply:
It's going to be difficult to avoid doing some kind of trick to continue to use the URL query parsing. Otherwise we will need to write our own parsing algorithm, which I think is more error prone. I am open to suggestions on how to do this in a manageable fashion.

Reply:
tried to have a look. I do not understand why there is a need for the didact suffix.
If I remove the if block, the tests are still passing.

Reply:
abandoning to refactor it. it is effectively complicated code to understand as it seems to sometimes use didact prefix, sometimes not.
it sound so tnot be the same path used in extensionFunction.test.ts and commandHandler.test.ts which is used.

Reply:
It took me a bit to process what you were saying here, but eventually I figured it out. Thanks. Have corrected the problems, added the additional test, and updated the example.